### PR TITLE
GCW-3101 Package sets

### PR DESCRIPTION
**Pull request by patrixr**
Changes:

* New `PackageSet` model/controller
* Removal of `set_item_id` on packages
* Addition of `package_set_id` on packages

Lifecycle:

* Packages created with the same parent item (from donor) are automatically added to a common set
* When a package is removed from a set (or destroyed), if a set is found with 1 item or less, it is destroyed

### CODE REVIEW

File: app/models/package_set.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  private
+
+  def unlink_packages
+    Package.where(package_set_id: id).update_all(package_set_id: nil)
+  end
+
+  def ensure_type_integrity
+    errors.add(:errors, I18n.t('package_sets.cannot_change_type')) if package_type_id_changed? && packages.length.positive?
```
Comment (steveyken) [inline app/models/package_set.rb:29]:
```suggestion
    errors.add(:errors, I18n.t('package_sets.cannot_change_type')) if package_type_id_changed? && packages.size.positive?
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  resources :package_sets, except: [:new, :edit]
```
Comment (steveyken) [inline config/routes.rb:2]: this is outside the `/api/v1/` scope. is this intended?

File: db/migrate/20200423034154_create_package_sets.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def change
+    create_table :package_sets do |t|
+      t.integer :package_type_id
+      t.text :description
+
+      t.timestamps null: false
+    end
+
+    add_column :packages, :package_set_id, :integer
+    remove_column :packages, :set_item_id, :integer
```
Comment (steveyken) [inline db/migrate/20200423034154_create_package_sets.rb:11]: Is 'set_item_id' column required for migration or just using item relationship?

File: config/routes.rb (hunk 2 above)
Comment (patrixr) [reply config/routes.rb:2]: oh, I used `rails generate scaffold` that's why. Will fix it

File: db/migrate/20200423034154_create_package_sets.rb (hunk 3 above)
Comment (patrixr) [reply db/migrate/20200423034154_create_package_sets.rb:11]: Yes, using the normal item relationship. Not needed

**Pull request merged**